Shared/Samples/Find route in mobile map package/FindRouteInMobileMapPackageView.MobileMapView.swift

```diff
@@ -0,0 +1,163 @@
+        @State private var errorAlertIsShowing = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error? {
+            didSet { errorAlertIsShowing = error != nil }
+        }
+        
+        init(map: Map, locatorTask: LocatorTask) {
+            let model = Model(map: map, locatorTask: locatorTask)
+            _model = StateObject(wrappedValue: model)
+        }
+        
+        var body: some View {
+            MapViewReader { mapViewProxy in
+                MapView(map: model.map, graphicsOverlays: model.graphicsOverlays)
+                    .callout(placement: $calloutPlacement) { _ in
+                        Text(calloutText)
+                            .font(.callout)
+                            .padding(8)
+                    }
+                    .onSingleTapGesture { screenPoint, _ in
+                        tapScreenPoint = screenPoint
+                    }
+                    .task(id: tapScreenPoint) {
+                        guard let tapScreenPoint else { return }
+                        
+                        do {
+                            // Check to see if the tap was on a marker.
+                            let identifyResult = try await mapViewProxy.identify(
+                                on: model.markerGraphicsOverlay,
+                                screenPoint: tapScreenPoint,
+                                tolerance: 12
+                            )
+                            
+                            if let graphic = identifyResult.graphics.first,
+                               let graphicPoint = graphic.geometry as? Point {
+                                // Update the callout to the identified marker.
+                                await updateCallout(point: graphicPoint, graphic: graphic)
+                            } else {
+                                // Add a graphic at the tapped map point.
+                                guard let location = mapViewProxy.location(
+                                    fromScreenPoint: tapScreenPoint
+                                ) else { return }
+                                await addGraphic(at: location)
+                            }
+                        } catch {
+                            self.error = error
+                        }
+                    }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Spacer()
+                    Button("Reset") {
+                        resetGraphics()
+                    }
+                    .disabled(resetDisabled)
+                }
+            }
+            .task {
+                // Load the default route parameters from the route task when the sample loads.
+                do {
+                    model.routeParameters = try await model.routeTask?.makeDefaultParameters()
```

Comment:
Should this happen inside the model? This requires a fair bit of information about model internals.